code-studio/src/js/components/progress/stage_progress.jsx

```diff
@@ -37,17 +46,32 @@ var StageProgress = React.createClass({
         }
       }
 
-      return ([
-        <div className={outerClass}>
-          <a
-            href={level.url}
-            className={innerClass + ' level-' + level.id}
-            style={dotStyle}>
-            {level.title}
-          </a>
-        </div>,
-        ' '
-      ]);
+      if (this.props.saveAnswersFirst) {
+        dotStyle.cursor = 'pointer';
+        return ([
+          <div className={outerClass}>
+            <span
+              onClick={()=>{this.dotClicked(level.url);}}
```

Comment:
Super small nit: spaces around `=>` and the extra `{...}` isn't needed for only one statement.
```javascript
onClick={() => this.dotClicked(level.url)}
```